execute/report/builder.go

```diff
@@ -176,55 +176,165 @@ func (b *execReportBuilder) checkInitialize() {
 // The commit report with updated metadata is returned. It reflects which messages
 // were selected for the exec report.
 //
-// TODO: Finish this function to generate multiple reports..
+// TODO: Add support for multiple reports with ordered messages.
 func (b *execReportBuilder) Add(
 	ctx context.Context,
 	commitReport exectypes.CommitData,
 ) (exectypes.CommitData, error) {
 	b.checkInitialize()
-	index := len(b.execReports) - 1
 
-	// Check if max is reached.
-	if b.maxSingleChainReports != 0 && len(b.execReports[index].ChainReports) >= int(b.maxSingleChainReports) {
-		if !b.multipleReportsEnabled {
-			// TODO: this is the previous behavior. Should we return an empty report error?
-			return commitReport, nil
-		} /* else {
-			// TODO: start a new exec report.
-		}*/
+	// Validate nonces for multiple reports mode
+	if err := b.validateNoncesForMultipleReports(commitReport); err != nil {
+		return commitReport, err
```

Comment:
This is a sidenote but this function's signature doesn't seem to make much sense to me. We return a commit report, but far as I can tell, its not really used. We seem to update a slice in-place with it but that slice isn't used (see https://github.com/smartcontractkit/chainlink-ccip/blob/7d0894c3f66876664301f8d63e938422212c3906/execute/plugin.go#L493-L498)

The `return` statements in this function end up returning a commit report even when an error is found; Go practice dictates that for errors we shouldn't return a usable object, otherwise its not really an error. Anyway, I feel like it'll be simpler if we just don't return the commit report here.